src/vs/workbench/contrib/debug/browser/debugService.ts

```diff
@@ -821,7 +821,8 @@ export class DebugService implements IDebugService {
 
 	private async showError(message: string, errorActions: ReadonlyArray<IAction> = []): Promise<void> {
 		const configureAction = new Action(DEBUG_CONFIGURE_COMMAND_ID, DEBUG_CONFIGURE_LABEL, undefined, true, () => this.commandService.executeCommand(DEBUG_CONFIGURE_COMMAND_ID));
-		const actions = [...errorActions, configureAction];
+		// Don't append the standard command if id of any provided action indicates it is a command
+		const actions = errorActions.filter((action) => action.id.endsWith('.command')).length > 0 ? errorActions : [...errorActions, configureAction];
```

Comment:
Do we have to check here if it `endsWith`. That feels a bit hacky?
Maybe just if it contains the word `command`?

Or even better can we use the `ICommandRegistry` to check if the `getCommand` exists for that `id`. That way we will know if the command is good or not?

Reply:
This is a convention I devised so handleErrorResponse in rawDebugSession.ts can indicate that the Action it supplied (currently only one) invokes a command. Here in showError I don't think there's any way I can discover this directly from the Action.

Another idea would be for handleErrorResponse to set the id of the Action to 'command:' followed by the id of the command (i.e. uri.authority), then make this check `startsWith('command:')`. It could even use `match` to extract the command id, then validate using `ICommandRegistry`. In that case, what's best to do if it fails validation? Omit the action from the messagebox? Or add it but disabled? Or don't suppress the DEBUG_CONFIGURE_COMMAND_ID?

What do you think?

Reply:
I think the current approach makes sense, and that your second proposal is very similar but just might complicate the implementation a bit. Due to that I suggest to go with the current approach.